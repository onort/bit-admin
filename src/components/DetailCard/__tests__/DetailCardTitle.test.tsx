import React from "react"
import { shallow } from "enzyme"

import { DetailCardTitle } from "../"

const testCardTitle = "TestCardTitle"
const testClassName = "testClassName"

describe("<DetailCardTitle />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<DetailCardTitle content={testCardTitle} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should display passed content", () => {
    const wrapper = shallow(<DetailCardTitle content={testCardTitle} />)
    expect(wrapper.find(".title").text()).toBe(testCardTitle)
  })

  it("should be able to have custom className", () => {
    const wrapper = shallow(
      <DetailCardTitle className={testClassName} content={testCardTitle} />
    )
    expect(wrapper.find(`.${testClassName}`).exists()).toBe(true)
  })
})
