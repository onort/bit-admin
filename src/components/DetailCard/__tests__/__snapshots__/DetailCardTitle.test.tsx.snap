// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCardTitle /> should match snapshot 1`] = `
<h2
  className="title"
>
  TestCardTitle
</h2>
`;

exports[`<DetailCardTitle /> should match snapshot 2`] = `"<h2 class=\\"title\\">TestCardTitle</h2>"`;
